src/core/aws-utils.adb

```diff
@@ -883,34 +887,64 @@ package body AWS.Utils is
    -- Random --
    ------------
 
-   function Random return Random_Integer is
+   function Random (Task_Safe : Boolean := False) return Random_Integer is
    begin
-      return Integer_Random.Random (Random_Generator);
+      return (if Task_Safe
+              then Random_Protecter.Random
+              else Integer_Random.Random (Random_Generator));
    end Random;
 
+   ----------------------
+   -- Random_Protecter --
+   ----------------------
+
+   protected body Random_Protecter is
```

Comment:
I would rename this Shared_Random as we have Shared_Sock in AWS.

Reply:
Done